app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -268,4 +259,41 @@ class MovementsController @Inject() (
         )
     }
 
+  private def ensureMovement(movement: Option[MovementWithoutMessages], movementId: MovementId): EitherT[Future, PresentationError, MovementWithoutMessages] =
+    EitherT(
+      Future.successful(movement.map(Right.apply).getOrElse(Left(PresentationError.notFoundError(s"Movement with ID ${movementId.value} was not found"))))
+    )
+
+  // Large Messages: Only perform this step if we haven't updated the movement EORI and we have an Object Store reference
+  private def updateMetadataIfRequired(
+    movementId: MovementId,
+    movementType: MovementType,
+    movementToUpdate: MovementWithoutMessages,
+    objectStoreResourceLocation: Option[ObjectStoreResourceLocation]
+  )(implicit hc: HeaderCarrier): EitherT[Future, PresentationError, Unit] =
+    (movementToUpdate.movementEORINumber, objectStoreResourceLocation) match {
+      case (None, Some(location)) =>
+        for {
+          source        <- objectStoreService.getObjectStoreFile(location).asPresentation
```

Comment:
``` objectStoreResourceLocation ``` here this will contain owner name ``` common-transit-convention-traders ``` with the passed object store uri and as we are using object store client we need to remove ``` common-transit-convention-traders ``` this part from the object store uri in order to match the file location with object store.

Reply:
Interesting. I looked through what was going on and while yes, you're right, this means that in a previous PR, `ObjectStoreResourceLocation` shouldn't have been used in the Json, so I'm gonna have to go fix that too.